### PR TITLE
Merging changes from the Summer 2019 ADA Demo

### PR DESCRIPTION
**Pull request by egordon**
Remove GCC7 / Ubuntu 18.04 compiler warnings, fixed NominalConfigurationRanker for ADA.

Tested on ADA directly, plus simulator tested via the feeding demo.

@gilwoolee Did you run any other tests?

***

**Before creating a pull request**

- [ ] Document new methods and classes
- [x] Format code with `make format`

**Before merging a pull request**

- [ ] Set version target by selecting a milestone on the right side
- [ ] Summarize this change in `CHANGELOG.md`
- [ ] Add unit test(s) for this change

### CODE REVIEW

File: include/aikido/constraint/FiniteSampleable.hpp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #ifndef AIKIDO_CONSTRAINT_FINITESAMPLEABLE_HPP_
 #define AIKIDO_CONSTRAINT_FINITESAMPLEABLE_HPP_
 
+#include "aikido/statespace/dart/MetaSkeletonStateSpace.hpp"
```
Comment (brianhou) [inline include/aikido/constraint/FiniteSampleable.hpp]: I don't think these change are right? They might also be duplicated from #529.

Comment (egordon) [reply include/aikido/constraint/FiniteSampleable.hpp]: Yeah, my mistake, let me roll back the master merge and re-do it.

**Comment by brianhou** [comment]
Didn't we just add `aikido::common::make_unique` in #532? I think that was perhaps so we could continue to compile on C++11?

**Comment by egordon** [comment]
@brianhou Yep, just noticed that, so I'm re-merging the change into this branch.

**Comment by egordon** [comment]
@brianhou Ok, ready for comment now I think.

File: src/constraint/dart/CollisionFree.cpp [hunk 2]
```diff
@@ -35,15 +35,22 @@ statespace::ConstStateSpacePtr CollisionFree::getStateSpace() const
 //==============================================================================
 bool CollisionFree::isSatisfied(
     const aikido::statespace::StateSpace::State* _state,
-    TestableOutcome* outcome) const
+    __attribute__((unused)) TestableOutcome* outcome) const
 {
+  // CollisionFreeOutcome* collisionFreeOutcome
```
Comment (brianhou) [inline src/constraint/dart/CollisionFree.cpp]: Should this just be deleted?

File: src/constraint/dart/CollisionFree.cpp [hunk 3]
```diff
@@ -35,15 +35,22 @@ statespace::ConstStateSpacePtr CollisionFree::getStateSpace() const
 //==============================================================================
 bool CollisionFree::isSatisfied(
     const aikido::statespace::StateSpace::State* _state,
-    TestableOutcome* outcome) const
+    __attribute__((unused)) TestableOutcome* outcome) const
```
Comment (brianhou) [inline src/constraint/dart/CollisionFree.cpp]: Should this just be deleted (or at least the parameter name, if we're not using it)?

File: src/constraint/dart/CollisionFree.cpp [hunk 4]
```diff
@@ -65,6 +72,7 @@ bool CollisionFree::isSatisfied(
       {
         collisionFreeOutcome->mPairwiseContacts = collisionResult.getContacts();
       }
+      std::cout << collisionFreeOutcome->toString() << std::endl;
```
Comment (brianhou) [inline src/constraint/dart/CollisionFree.cpp]: Is this for debugging?

File: src/constraint/dart/InverseKinematicsSampleable.cpp [hunk 5]
```diff
@@ -220,7 +220,8 @@ bool IkSampleGenerator::sample(statespace::StateSpace::State* _state)
 
     mInverseKinematics->getTarget()->setTransform(poseState.getIsometry());
 
-    // Run the IK solver. If an exact solution is computed, apply it to the skeleton.
+// Run the IK solver. If an exact solution is computed, apply it to the
```
Comment (brianhou) [inline src/constraint/dart/InverseKinematicsSampleable.cpp]: Can we move this comment inside the `#if`? Would that help the formatter put it at the right indentation level?

File: src/control/KinematicSimulationTrajectoryExecutor.cpp [hunk 6]
```diff
@@ -139,6 +139,7 @@ void KinematicSimulationTrajectoryExecutor::step(
   // Check if trajectory has completed.
   if (executionTime >= mTraj->getEndTime())
   {
+    std::cout << "Trajectory completed" << std::endl;
```
Comment (brianhou) [inline src/control/KinematicSimulationTrajectoryExecutor.cpp]: Remove?

File: src/control/ros/Conversions.cpp [hunk 7]
```diff
@@ -265,8 +276,29 @@ void extractTrajectoryPoint(
...
+  auto geodesicInterpolator
+      = std::dynamic_pointer_cast<aikido::statespace::GeodesicInterpolator>(
+          interpolator);
+  if (!geodesicInterpolator)
+  {
+    throw std::invalid_argument(
+        "The interpolator of trajectory should be a GeodesicInterpolator");
+  }
+
+  auto diff = geodesicInterpolator->getTangentVector(prevState, state);
```
Comment (brianhou) [inline src/control/ros/Conversions.cpp]: Hmm, this seems to all be part of the SO(2) and R1 conversion that was originally part of #496. [AVK and I decided at the time](https://github.com/personalrobotics/aikido/pull/496#issuecomment-463749864) to address this in rewd_controllers instead (https://github.com/personalrobotics/rewd_controllers/issues/30). Has that changed?

File: src/control/ros/RosTrajectoryExecutor.cpp [hunk 8]
```diff
@@ -150,6 +150,7 @@ std::future<void> RosTrajectoryExecutor::execute(
         goal,
         boost::bind(&RosTrajectoryExecutor::transitionCallback, this, _1));
 
+    std::cout << "Returning future" << std::endl;
```
Comment (brianhou) [inline src/control/ros/RosTrajectoryExecutor.cpp]: Delete?

File: src/planner/SnapConfigurationToConfigurationPlanner.cpp [hunk 9]
```diff
@@ -37,6 +38,7 @@ trajectory::TrajectoryPtr SnapConfigurationToConfigurationPlanner::plan(
   auto constraint = problem.getConstraint();
 
   aikido::common::VanDerCorput vdc{1, true, true, 0.02}; // TODO junk resolution
+
```
Comment (brianhou) [inline src/planner/SnapConfigurationToConfigurationPlanner.cpp]: Delete?

File: src/rviz/ShapeFrameMarker.cpp [hunk 10]
```diff
@@ -39,7 +39,7 @@ ShapeFrameMarker::ShapeFrameMarker(
   , mForceUpdate(true)
   , mVersion()
   , mShowVisual(true)
-  , mShowCollision(false)
+  , mShowCollision(true)
```
Comment (brianhou) [inline src/rviz/ShapeFrameMarker.cpp]: Why change this default?

File: src/trajectory/util.cpp [hunk 11]
```diff
@@ -151,52 +155,60 @@ UniqueInterpolatedPtr concatenate(
 //==============================================================================
 double findTimeOfClosestStateOnTrajectory(
```
Comment (brianhou) [inline src/trajectory/util.cpp]: Hmm, should this function remain in `aikido::trajectory` or would it make sense to have it in `aikido::distance` instead?

File: src/trajectory/CMakeLists.txt [hunk 12]
```diff
@@ -9,6 +9,7 @@ add_library("${PROJECT_NAME}_trajectory" SHARED ${sources})
 target_link_libraries("${PROJECT_NAME}_trajectory"
   PUBLIC
     "${PROJECT_NAME}_common"
+    "${PROJECT_NAME}_distance"
```
Comment (brianhou) [inline src/trajectory/CMakeLists.txt]: If we keep this dependency, we should make sure to update the comments of `src/CMakeLists.txt` too.

File: src/trajectory/util.cpp [hunk 13]
```diff
@@ -151,52 +155,60 @@ UniqueInterpolatedPtr concatenate(
...
 
+  distance = minDist;
   return findTime;
 }
 
 //==============================================================================
 UniqueSplinePtr createPartialTrajectory(
     const Spline& traj, double partialStartTime)
 {
+  std::cout << "traj starts at " << traj.getStartTime() << " ends at "
```
Comment (brianhou) [inline src/trajectory/util.cpp]: Delete?

File: src/trajectory/util.cpp [hunk 14]
```diff
@@ -151,52 +155,60 @@ UniqueInterpolatedPtr concatenate(
...
+            << traj.getEndTime() << std::endl;
   if (partialStartTime < traj.getStartTime()
       || partialStartTime > traj.getEndTime())
   {
-    throw std::runtime_error("Wrong partial start time");
+    dtwarn << "Wrong partial start time" << std::endl;
```
Comment (brianhou) [inline src/trajectory/util.cpp]: Why warn instead of error?

File: src/trajectory/util.cpp [hunk 15]
```diff
@@ -254,6 +266,8 @@ UniqueSplinePtr createPartialTrajectory(
         traj.getSegmentStartState(i));
   }
 
+  std::cout << "outputTrajectory starts at " << outputTrajectory->getStartTime()
```
Comment (brianhou) [inline src/trajectory/util.cpp]: Delete?

File: src/control/ros/Conversions.cpp (hunk 7 above)
Comment (egordon) [reply src/control/ros/Conversions.cpp]: @brianhou It probably hasn't. Some of these changes may be holdovers from an earlier branch. I'm actually considering completely abandoning this branch, as it seems a lot of the changes are redundant.

**Comment by egordon** [comment]
Rendered out-dated by #543

**Pull request closed without merging**